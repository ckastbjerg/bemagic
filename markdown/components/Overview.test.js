/* eslint-env jest */
const Overview = require('./Overview');

test('it throws', () => {
    expect(() => Overview()).toThrow();
});

test('it works', () => {
    expect(Overview({
        tag: 'div',
        name: 'block',
        states: ['is-dimmed'],
        elements: {
            element: {
                tag: 'div',
                name: 'element',
                states: ['is-dimmed'],
                modifiers: {
                    modifier: {
                        tag: 'div',
                        name: 'modifier',
                        states: ['is-something'],
                    },
                },
            },
        },
        modifiers: {
            modifier: {
                tag: 'div',
                name: 'modifier',
                states: ['is-something'],
            },
        },
    })).toMatchSnapshot();
});
