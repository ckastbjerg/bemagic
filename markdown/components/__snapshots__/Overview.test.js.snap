// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it works 1`] = `
"
        <a name=Component Overview></a>
        #### Component Overview
    | Class Name | Type | States | Tag |
|:-----|:-----|:-----|:-----|
| **[\`block\`](#block)** | [component](../CONVENTIONS.md#components) | \`is-dimmed\`. | N/A |
| **[\`modifier\`](#modifier)** | [modifier](../CONVENTIONS.md#modifiers) | \`is-something\`. | N/A |
| **[\`element\`](#element)** | [element](../CONVENTIONS.md#elements) | \`is-dimmed\`. | N/A |
| **[\`modifier\`](#modifier)** | [modifier](../CONVENTIONS.md#modifiers) | \`is-something\`. | N/A |


"
`;
